Src/Support/Google.Apis.Auth.Tests/GoogleJsonWebSignatureTests.cs

```diff
@@ -86,52 +88,101 @@ public class GoogleJsonWebSignatureTests
+            new GoogleJsonWebSignature.ValidationSettings
+            {
+                Clock = clock,
+                Audience = aud == null ? null : new[] { aud },
+                HostedDomain = hd,
+            };
+
+        private string B64(string t) =>
+            Convert.ToBase64String(Encoding.UTF8.GetBytes(t)).Replace("=", "");
+
+        [Fact]
+        public async Task Validate_BadJwt()
+        {
+            var settings = S(new MockClock() { UtcNow = new DateTime(2017, 5, 31, 10, 22, 0, DateTimeKind.Utc) });
+            // Null JWT
+            await Assert.ThrowsAsync<ArgumentNullException>(() => GoogleJsonWebSignature.ValidateInternalAsync(null, settings, GoogleCertsJson, false));
+            // Empty JWT
+            await Assert.ThrowsAsync<ArgumentException>(() => GoogleJsonWebSignature.ValidateInternalAsync("", settings, GoogleCertsJson, false));
+            // Too long JWT
+            await Assert.ThrowsAsync<InvalidJwtException>(() =>
+                GoogleJsonWebSignature.ValidateInternalAsync(new string('a', GoogleJsonWebSignature.MaxJwtLength + 1), settings, GoogleCertsJson, false));
+            // JWT with incorrect top-level structure; missing signature
+            await Assert.ThrowsAsync<InvalidJwtException>(() =>
+                GoogleJsonWebSignature.ValidateInternalAsync("header.payload", settings, GoogleCertsJson, false));
+        }
+
+        [Fact]
+        public async Task WrongAlgorithm()
+        {
+            var settings = S(new MockClock() { UtcNow = new DateTime(2017, 5, 31, 10, 22, 0, DateTimeKind.Utc) });
+            string jwt = $"{B64("{\"alg\":\"HS256\"}")}.{B64("{}")}.";
+            var ex = await Assert.ThrowsAsync<InvalidJwtException>(() =>
+                GoogleJsonWebSignature.ValidateInternalAsync(jwt, settings, null, true));
+            Assert.Equal("JWT algorithm must be 'RS256'", ex.Message);
+        }
+
         [Fact]
         public async Task Validate_Signature()
         {
             var clockValid1 = new MockClock() { UtcNow = new DateTime(2017, 5, 31, 10, 24, 0, DateTimeKind.Utc) };
-            Assert.NotNull(await GoogleJsonWebSignature.ValidateInternalAsync(JwtGoogleSigned, clockValid1, false, GoogleCertsJson));
+            Assert.NotNull(await GoogleJsonWebSignature.ValidateInternalAsync(JwtGoogleSigned, S(clockValid1), GoogleCertsJson, false));
 
             var clockValid2 = new MockClock() { UtcNow = new DateTime(2017, 5, 31, 10, 24, 0, DateTimeKind.Utc) };
             var ex = await Assert.ThrowsAsync<InvalidJwtException>(() =>
-                GoogleJsonWebSignature.ValidateInternalAsync(JwtNonGoogleSigned, clockValid2, false, GoogleCertsJson));
-            Assert.Equal("JWT invalid: unable to verify signature.", ex.Message);
+                GoogleJsonWebSignature.ValidateInternalAsync(JwtNonGoogleSigned, S(clockValid2), GoogleCertsJson, false));
```

Comment:
Is "S" already defined somewhere? Assuming it's quoting the string, I'd prefer the name "Quote".